cdap-docs/developers-manual/source/building-blocks/workflows.rst

```diff
@@ -86,12 +94,206 @@ implement the ``run()`` method::
+use  multiple instances of the same program in the same workflow.
+
+These unique names can be set when the Workflow is first configured, passed to the instance of the
+program, and then be used when the program performs its own configuration. An example
+of this is the :ref:`Wikipedia Pipeline<wikipedia-data-pipeline>` example, and its use of the
+*StreamToDataset* MapReduce program multiple times::
+
+  public class StreamToDataset extends AbstractMapReduce {
+    private static final Logger LOG = LoggerFactory.getLogger(StreamToDataset.class);
+  
+    private final String name;
+  
+    public StreamToDataset(String name) {
+      this.name = name;
+    }
+  
+    @Override
+    public void configure() {
+      setName(name);
+      ...
+    }
+
+In its declaration of the application, the example specifies the unique names for each instance::
+
+  public class WikipediaPipelineApp extends AbstractApplication {
+    ...
+    static final String LIKES_TO_DATASET_MR_NAME = "likesToDataset";
+    static final String WIKIPEDIA_TO_DATASET_MR_NAME = "wikiDataToDataset";
+
+    @Override
+    public void configure() {
+      ...
+      addMapReduce(new StreamToDataset(LIKES_TO_DATASET_MR_NAME));
+      addMapReduce(new StreamToDataset(WIKIPEDIA_TO_DATASET_MR_NAME));
+      ...
+      addWorkflow(new WikipediaPipelineWorkflow());
+    }
+  }
+
+The workflow itself uses the same names in its configuration::
+
+  public class WikipediaPipelineWorkflow extends AbstractWorkflow {
+
+    public static final String NAME = WikipediaPipelineWorkflow.class.getSimpleName();
+
+    @Override
+    protected void configure() {
+      setName(NAME);
+      setDescription("A workflow that demonstrates a typical data pipeline to process Wikipedia data.");
+      addMapReduce(WikipediaPipelineApp.LIKES_TO_DATASET_MR_NAME);
+      condition(new EnoughDataToProceed())
+        .condition(new IsWikipediaSourceOnline())
+          .addAction(new DownloadWikiDataAction())
+        .otherwise()
+          .addMapReduce(WikipediaPipelineApp.WIKIPEDIA_TO_DATASET_MR_NAME)
+        .end()
+        .addMapReduce(WikiContentValidatorAndNormalizer.NAME)
+        .fork()
+          .addSpark(SparkWikipediaAnalyzer.NAME)
+        .also()
+          .addMapReduce(TopNMapReduce.NAME)
+        .join()
+      .otherwise()
+      .end();
+    }
+
+.. _workflow-concurrent:
+
 Concurrent Workflows
 --------------------
 By default, a workflow runs sequentially. Multiple instances of a workflow can be run
 concurrently. To enable concurrent runs for a workflow, set its runtime argument
 ``concurrent.runs.enabled`` to ``true``.
 
+
+.. _workflow_token:
+
+Workflow Tokens
+===============
+
+In addition to passing the control flow from one node to the next, a **workflow token** is
+passed, available to each of the programs in the workflow. This allows programs to:
+
+- pass custom data (such as metrics, a status, or an error code) from one program in the 
```

Comment:
Could a user write their own code to read or write to metrics, and pass the values or names along? If so, I would leave it in, because that is how I would interpret it.